specs/pkg-js/trustless-pathing/testcase.js

```diff
@@ -0,0 +1,82 @@
+import { CID } from 'multiformats/cid'
+
+class TestCase {
+  constructor (name, root, path, scope, duplicates, byteRange, expectedCids) {
+    this.name = name
+    this.root = root
+    this.path = path
+    this.scope = scope
+    this.duplicates = duplicates
+    this.byteRange = byteRange
+    this.expectedCids = expectedCids
+  }
+
+  asQuery () {
+    let pp = this.path.split('/').filter(Boolean).join('/')
+    if (pp !== '') {
+      pp = '/' + pp
+    }
+    const br = this.byteRange ? `&entity-bytes=${this.byteRange}` : ''
+    const dup = this.duplicates ? '&dups=y' : ''
+    return `/ipfs/${this.root}${pp}?dag-scope=${this.scope}${br}${dup}`
+  }
+}
+
+export function parseCase (name, spec, exec) {
+  let lines = exec.split('\n')
+  while (lines.length > 0 && lines[0].trim() === '') {
+    lines = lines.slice(1)
+  }
+  while (lines.length > 0 && lines[lines.length - 1].trim() === '') {
+    lines = lines.slice(0, lines.length - 1)
+  }
+  const specParts = spec.trim().split('?')
+  if (specParts.length !== 2) {
+    return new Error('invalid spec')
```

Comment:
Did you mean to return these? Should be throw? Note: multiple places.

Reply:
go-brain probably, banged this out pretty quick